xdsl/dialects/builtin.py

```diff
@@ -1725,132 +1726,61 @@ def get_element_type(self) -> _UnrankedMemrefTypeElems:
 @irdl_attr_definition
 class DenseIntOrFPElementsAttr(TypedAttribute, ContainerType[AnyDenseElement]):
     name = "dense"
-    type: ParameterDef[
-        RankedStructure[IntegerType]
-        | RankedStructure[IndexType]
-        | RankedStructure[AnyFloat]
-    ]
-    data: ParameterDef[ArrayAttr[AnyIntegerAttr] | ArrayAttr[AnyFloatAttr]]
+    type: ParameterDef[RankedStructure[IntegerType | IndexType | AnyFloat]]
+    data: ParameterDef[BytesAttr]
```

Comment:
I would recommend first migrating all users of this property to another property/getter that returns a sequence of AnyIntegerAttr or a sequence of AnyFloatAttr, to minimise the diff in other files (probably also a property/getter that returns a sequence of ints or a sequence of floats)